### ISSUE TITLE
Add support for other platforms beyond WordPress

### ISSUE THREAD

**Issue by EHLOVader** [Varying-Vagrant-Vagrants/VVV#565]
I am investigating the possibility of other platforms like Drupal, Laravel etc. They each have similar command line managers and I wanted to check if there was a community need or others who would like it.

I have gotten other sites to work with the standard site wizard configuration and I think that it might be nice to have. Any others interested?

**Comment by cfoellmann** [comment]
I think VVV should stay focused on WordPress. On modern machines you can easily run multiple VMs so it might be best use VVV for WP and different specialized Vagrants for other environments

**Comment by EHLOVader** [comment]
Thanks for the reply!

I can understand the reservations though. I don't feel that anything has to change within VVV. Using multiple virtual machines is not an option for me either. Imagine if you designed VVV to use separate machines per site (perhaps that is where it started) how much more complex and complicated it would be, not to mention cumbersome. 

I mostly wanted to find out if others had interest, but the responsibility for making it work would be outside of this project. I have already gotten [Drupal](https://github.com/EHLOVader/vvv-drupal) working with great success.

**Comment by cfoellmann** [comment]
Running other PHP stuff on VVV is not the problem like you already stated but VVV is more than just another PHP Vagrant box.
You can create auto setup site scripts for all the other stuff but adding equivalents to the tools for other platforms would be too far out of scope.

**Comment by EHLOVader** [comment]
Take a peek at my Drupal repo I linked. You will see there is no need to modify core to install Drush and use it for configuring the site. 

Again I guess this was mostly to guage interest, from community and the project maintainers. 

Your feedback has been very helpful.

**Comment by cfoellmann** [comment]
Ok. Happy to be of help.
Your autoconfig is nice. What else would you like to add? 

Or would you like to move it to the VVV org? To make it more visible? We could also add a page to the wiki.

**Comment by EHLOVader** [comment]
For Laravel and any random PHP project I think that Composer is all those types of projects would need, which is already present or in the works.

The main thing with these, if I do find a way to add many is finding a way to organize it without interfering with the projects themselves. I definitely think my next step on these is to find a way to keep the supporting files out of the www folder if at all possible.

**Comment by jeremyfelt** [comment]
Our general objective is to provide a good environment for WordPress development that matches a good production environment. While we'll likely not include support for things like Drupal directly, anything is possible with VVV init scripts.

Once we have config files hammered out, this will be even easier to add on individually.

**Comment by EHLOVader** [comment]
Perfectly clear, thanks @jeremyfelt. I agree that VVV doesn't need any changes to it for supporting other platforms. 

Wanted to know if there was interest though, and perhaps others already doing this locally from whom I could learn. The VVV init scripts have helped wonderfully. 

Tell me more about these configs though? Is this from the provision splitting issue? #319

**Comment by jeremyfelt** [comment]
@EHLOVader It goes along with the provision splitting issue. #474 is probably the most clearly stated, and #210 goes hand in hand.

**Comment by EHLOVader** [comment]
Awesome. Thanks.

I know docker may be in the future also, which could help with this. But
was another provisioner ever considered? Like Ansible, puppet, chef, etc?

**Comment by bradp** [comment]
I just want to let you know that I've gotten laravel and other non-WP installs to work without an issue inside of VVV using [vv](https://github.com/bradp/vv)'s --blank sites.

There's nothing that needs to be done on the VVV side, other than installing any command line tools you'd want, like artisan or drush, but those can be done in a custom provision script.

**Comment by EHLOVader** [comment]
Thanks for the reply @bradp If you could, please re read my original post and the followup [comment in which](https://github.com/Varying-Vagrant-Vagrants/VVV/issues/565#issuecomment-74283781) I linked to a Drupal setup I have had success with.

Given the general tone of responses I seem to have failed to convey that I wasn't suggesting changes to VVV as I stated it is already possible to use other php platforms within it, and that I already had. But rather that I was reaching out to the VVV community for input on need, and successes or pain points (like where to put the additional tools you are provisioning sensibly). There isn't a Freenode channel or other means to communicate with the community outside of here that I know of.

I appreciate the feedback and definitely think I will pursue at least a library of these recipes to use other platforms. But feel that this thread is not constructive any longer as the intention has been missed. I am closing. 

Thank you.

**Comment by jeremyfelt** [comment]
@EHLOVader Finally following up on a question you asked previously. :smile: We did go down the path of a proper provisioner at one point in #65 and bailed. I'm still mixed personally as I've embraced Salt for other provisioning projects. Not ruled out completely, but we'd need a nice plan going in. Part of me enjoys the clarity that `provision.sh` offers a newcomer to the project, part of me wonders if anyone really wants to look at that.

Also a general +1 on finding ways to use VVV for other projects. I think the base nginx, mysql, php setup can fit a lot of use cases, even if some of the software provisioned would be extra.
